roles/docker/templates/docker-svc.j2

```diff
@@ -1,17 +1,27 @@
 [Unit]
 Description=Docker Application Container Engine
 Documentation=https://docs.docker.com
-After=network.target docker.socket
-Requires=docker.socket
+After=network.target
 
 [Service]
 TimeoutStartSec=5m
 Type=notify
 {% if docker_device != "" %}
-ExecStart=/usr/bin/docker daemon --storage-opt dm.thinpooldev=/dev/mapper/contiv-dockerthin -H fd:// --cluster-store=etcd://localhost:{{ etcd_client_port1 }}
+{% set docker_devmapper_opts = "--storage-opt dm.thinpooldev=/dev/mapper/contiv-dockerthin" %}
 {% else %}
-ExecStart=/usr/bin/docker daemon -H fd:// --cluster-store=etcd://localhost:{{ etcd_client_port1 }}
+{% set docker_devmapper_opts = "" %}
 {% endif %}
+
+{% if docker_version | version_compare('1.12', '>=') %}
+{% set docker_daemon_startup = "dockerd" %}
+{% else %}
+{% set docker_daemon_startup = "docker daemon" %}
```

Comment:
@DivyaVavili This is handled by lines 21 and 23 of this template. The UNIX socket and TCP socket options are passed directly to the Docker daemon as options on the command line. The docker-tcp.socket is no longer required.

It doesn't make sense to keep docker-tcp.socket for older versions. Older versions of Docker can also use this setup with `-H tcp://0.0.0.0:2375 -H unix:///var/run/docker.sock`. This was used to have socket activation and it was more trouble than it was worth. We can just use the latest recommended setup.

I've been using this setup for a long time. I haven't run into any kind of issue so far.